exercises/pythagorean-triplet/example.py

```diff
@@ -1,37 +1,61 @@
-def triplets_in_range(min, max):
-    result = set([x for b in
-                  range(4, max + 1, 4) for x in
-                  primitive_triplets(b)])
-    for x, y, z in set(result):
-        a, b, c = (2 * x, 2 * y, 2 * z)
-        while c <= max:
-            if a >= min:
-                result.add((a, b, c))
-            a, b, c = (a + x, b + y, c + z)
-    return set([(d, e, f) for d, e, f in result if d >= min and f <= max])
+from math import sqrt, ceil
 
 
-def maxMin(a, b):
-    return (a, b) if a > b else (b, a)
+def triplets_in_range(min, max):
+    result = []
+    for b in range(4, max + 1, 4):
+        for x, y, z in primitive_triplets(b):
+            a, b, c = (x, y, z)
+            while a < min:
+                a, b, c = (a + x, b + y, c + z)
+            while c <= max:
+                result.append((a, b, c))
+                a, b, c = (a + x, b + y, c + z)
+    return result
 
 
 def gcd(x, y):
```

Comment:
Since you're already using `math.ceil` and `math.sqrt`, why not `math.gcd` as well?

Reply:
Sure, works for me.